azkaban-web-server/src/main/java/cloudflow/daos/FlowDaoImpl.java

```diff
@@ -0,0 +1,118 @@
+package cloudflow.daos;
+
+import azkaban.db.DatabaseOperator;
+import cloudflow.error.CloudFlowNotFoundException;
+import cloudflow.models.FlowResponse;
+import cloudflow.models.Project;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import javax.inject.Inject;
+import org.apache.commons.dbutils.ResultSetHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FlowDaoImpl implements FlowDao {
+
+  private DatabaseOperator dbOperator;
+  private FlowAdminDaoImpl flowAdminDaoImpl;
+  private ProjectDao projectDao;
+  private static final Logger log = LoggerFactory.getLogger(ProjectDaoImpl.class);
+
+  @Inject
+  public FlowDaoImpl(DatabaseOperator dbOperator, FlowAdminDaoImpl flowAdminDaoImpl,
+      ProjectDao projectDao) {
+    this.dbOperator = dbOperator;
+    this.flowAdminDaoImpl = flowAdminDaoImpl;
+    this.projectDao = projectDao;
+  }
+
+
+  @Override
+  public Optional<FlowResponse> getFlow(String flowId){
+    List<FlowResponse> flows = new ArrayList<>();
+    FetchFlowHandler fetchFlowHandler = new FetchFlowHandler();
+    try {
+      flows = dbOperator.query(FetchFlowHandler.SELECT_FLOW_WITH_FLOW_ID,
+          fetchFlowHandler, flowId);
+      for(FlowResponse flow : flows) {
+        //flow.setAdmins(flowAdminDaoImpl.findAdminsByFlowId(flow.getId()));
```

Comment:
Currently the Python DSL does not provide a way to set flow admins. Once that feature is supported, this code will be useful. Since it is already implemented (and tested) I didn't want to take it out. Will add a comment why is it is commented.